src/neuron/container/soa_container.hpp

```diff
@@ -819,55 +896,94 @@ struct soa {
     /**
      * @brief Query if the underlying vectors are still "sorted".
      *
-     * See the documentation of get_sorted_token() for an explanation of what
+     * See the documentation of issue_frozen_token() for an explanation of what
      * this means.
      */
     [[nodiscard]] bool is_sorted() const {
         return m_sorted;
```

Comment:
I don't think adding a mutex lock here would achieve anything.
I believe it is safe to use this signature in code like:
```c++
auto token = soa_data.issue_frozen_token();
auto sorted = soa_data.is_sorted();
```
as the sorted-ness cannot change while >1 tokens are active, and we know that it's not changing because of the token in this snippet.